<launch>
  <arg name="x" default="0" />
  <arg name="y" default="0" />
  <arg name="z" default="0.4" />
  <arg name="yaw" default="0" />
  <arg name="world_name" default="$(find jackal_gazebo)/worlds/jackal_race_top_down.world" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="debug" value="0" />
    <arg name="gui" value="false" />
    <arg name="use_sim_time" value="true" />
    <arg name="headless" value="false" />
    <arg name="world_name" value="$(arg world_name)" />
  </include>


  <!-- Configuration of Jackal which you would like to simulate. See jackal_description for details. -->
  <!-- Spawn Jackal -->
  <include file="$(find jackal_gazebo)/launch/spawn_jackal.launch">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="$(arg z)" />
    <arg name="yaw" value="$(arg yaw)" />
    <arg name="config" value="front_laser" />
    <arg name="joystick" value="true" />
    <arg name="keyboard" default="false" />
    <arg name="enable_ekf" default="false" />
  </include>

  <node pkg="tf2_ros" type="static_transform_publisher" name="idendity_map" args="0 0 0 0 0 0 1 scanmatcher_frame base_link"/>

  <!-- slam hector as mapping and localization does not use external odom-->
  <include file="$(find jackal_gazebo)/launch/jackal_hector_map.launch">
    <arg name="scan_topic" value="/front/scan" />
    <arg name="map_size" value="255" />
    <arg name="map_resolution" value="0.1" />
    <arg name="map_frame" value="map" />
    <arg name="map_start_x" default="0.5"/>
    <arg name="map_start_y" default="0.5"/>
    <arg name="map_pub_period" value="0.02" />
  </include>

  <!--- Run costmap computation -->
  <node name="erl_costmap_ros" pkg="erl_costmap_ros" type="erl_costmap_ros_node" output="screen" launch-prefix="">
    <param name="gridmsg_obstacle_lower_bound" value="80"/>
    <param name="gridmsg_obstacle_upper_bound" value="100"/>
    <param name="gridmsg_unknown" value="-1"/>

    <param name="costmap_planning_robot_inscribed_radius" value="0.215"/>
    <param name="costmap_planning_gamma" value="7.0"/>
    <param name="costmap_planning_unknown" value="3"/>
    <param name="costmap_planning_lethal" value="19"/>
    <param name="costmap_planning_inscribed" value="19"/>
    <param name="costmap_planning_cutoff_cost" value="5"/>

    <param name="costmap_visualization_unknown" value="-1"/>
    <param name="costmap_visualization_lethal" value="100"/>
    <param name="costmap_visualization_inscribed" value="90"/>
    <param name="costmap_visualization_planning_cutoff" value="-3"/>
    <param name="costmap_visualization_zero_cost" value="110"/>

  </node>


  <!--- Run rviz using config inside package -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find erl_costmap_ros)/rviz/costmap.rviz" />

  <!-- Run multiplot-->
  

</launch>
